File: Beginner/tickets/tickets.c
#include <stdio.h>
#include <assert.h>
#include <string.h>

int main(void){
	int T,flag;

	scanf("%d",&T);
	assert(T>=1 && T<=100);

	while(T>0){
		char S[100];
		scanf("%s",S);

		for(int i=0;i<strlen(S);i++){
			assert(S[i]>=65 && S[i]<=90);
		}

		if(S[0]==S[1]){
			printf("NO\n");
		}else{
			flag = 0;

			for(int i=0;i<strlen(S);i++){
				if(i%2==0){
					if(S[0]!=S[i]){
						flag = 1;
						break;
					}
				}else{
					if(S[1]!=S[i]){
						flag = 1;
						break;
					}
				}
			}

			if(flag==0){
				printf("YES\n");
			}else{
				printf("NO\n");
			}
		}

		T--;
	}
	
	return 0;
}

File: Beginner/tickets/tickets.java
//run as java -ea tickets
import java.util.Scanner;

public class tickets{
	public static void main(String[] args){
		int T,flag;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=100) : "T must be 1 <= T <= 100";

		while(T>0){
			String S;
			S = scan.next();

			for(int i=0;i<S.length();i++){
				assert(S.charAt(i)>=65 && S.charAt(i)<=90) : "All letters must be upper-case latin letters";
			}

			if(S.charAt(0)==S.charAt(1)){
				System.out.println("NO");
			}else{
				flag = 0;

				for(int i=0;i<S.length();i++){
					if(i%2==0){
						if(S.charAt(0)!=S.charAt(i)){
							flag = 1;
							break;
						}
					}else{
						if(S.charAt(1)!=S.charAt(i)){
							flag = 1;
							break;
						}
					}
				}

				if(flag==0){
					System.out.println("YES");
				}else{
					System.out.println("NO");
				}
			}

			T--;
		}
	}
}